### DISCUSSION TITLE
Retrieving parameter

### DISCUSSION THREAD

**Discussion by farhanaazzz** [orchidsoftware/platform#1727] [Q&A]
I have a screen (view) containing a list of users in table form. There is a column for editing the user's details which I made in a popup form. I have a problem passing the parameter from the screen to the Blade popup form.  When I clicked on the modal popup, the information the user entered are not there eventhough I retrieved it using the 'old' function. This led me to believe that the user ID is not passed well from the PHP file. Here is the code for clicking the modal popup:

**UserDetailsScreen.php**

    //modal toggle in table
    TD::make('action', 'Action')
        ->render(function($user){
            return ModalToggle::make('Edit user')
              ->modal('editModal')
              ->modalTitle('Edit this user')

              //I pass the parameter using this
              ->asyncParameters([
                 'id' => $user->id
               ])
              ->class('edit-modal')
              ->method('saveUser');
       }

    //view modal popup view
    Layout::modal('editModal', [

                Layout::view('editForm'),
    ]),

When I inspect one of the modal popup, the id of the user did appear in the elements like this:

    data-screen--modal-toggle-action="http://127.0.0.1:8000/admin/userdetails/saveUser?id=18"

But not in the URL of the webpage since it is just a popup window.

Here is my custom blade form for the popup. The JavaScript part at the end is the part where I want to retrieve the parameter.

**editForm.blade.php**

    <!DOCTYPE html>
    <html>
        <body>
        <div>
            <div style = "padding-top: 30px; padding-left: 30px;">
                {!! 
                    \Orchid\Screen\Fields\Input::make('user.first_name')
                        ->title('First Name')
                        ->value(old('first_name'))
                        ->horizontal()
                !!}
            </div>
            <div style = "padding-left: 30px;">
                {!! 
                    \Orchid\Screen\Fields\Input::make('user.last_name')
                        ->title('Last Name')
                        ->value(old('last_name'))
                        ->horizontal()
                !!}
            </div>
            <div style = "padding-left: 30px;">
                {!! 
                    \Orchid\Screen\Fields\Input::make('user.address')
                        ->title('Address')
                        ->horizontal()
                !!}
        </div>
    
        <script>
             $(document).on("click", ".edit-modal", function() {
               //retrieve the parameter
            })
        </script>
        </body>
    </html>

I have tried retrieving the parameter using *.val()* but it does not work. Does anyone have any idea? Thanks.

**Comment by sangamu** [comment]
@farhanazulaikha  You can add one hidden field in your edit blade file.

Input::make('user.id') ->type('hidden');

You can inspect for name of this field in the generated form. I believe it will have randomly generated id but name of the field will be user[id]. From your jquery event, you can read the dom element with this name and get value.

Hope this helps.
